operator/src/main/resources/scripts/traceUtils.sh

```diff
@@ -164,3 +164,107 @@ function exportEffectiveDomainHome() {
   trace "Error: More than one config.xml found at DOMAIN_HOME/config/config.xml and DOMAIN_HOME/*/config/config.xml, DOMAIN_HOME='$DOMAIN_HOME': ${found_configs}. Configure your 'domainHome' setting in your WebLogic Operator Domain resource to reference a single WebLogic domain."
   return 1
 }
+
+
+# versionCmp
+#   Compares two wl versions $1 $2 up to the length of $2
+#     Expects form N.N.N.N
+#     Uses '0' for an N in $1 if $1 is shorter than $2
+#   echo "1"  if v1 >  v2
+#   echo "-1" if v1 <  v2
+#   echo "0"  if v1 == v2
+versionCmp()
+{
+  IFS='.' read -r -a v1_arr <<< "`echo $1`"
+  IFS='.' read -r -a v2_arr <<< "`echo $2`"
+
+  for i in "${!v2_arr[@]}"
+  do
+    [ ${v1_arr[i]:-0} -gt ${v2_arr[i]:-0} ] && echo "1" && return
+    [ ${v1_arr[i]:-0} -lt ${v2_arr[i]:-0} ] && echo "-1" && return
+  done
+  echo "0"
+}
+
+# versionGE
+#   return success if WL v1 >= v2
+versionGE()
+{
+  [ `versionCmp "$1" "$2"` -ge 0 ] && return 0
+  return 1
+}
+
+# versionEQ
+#   return success if v1 == v2
+versionEQ()
+{
+  [ `versionCmp "$1" "$2"` -eq 0 ] && return 0
+  return 1
+}
+
+# hasWebLogicPatches
+#   check for the given patch numbers in the install inventory, 
+#   and return 1 if not found
+#   - if we can't find the install inventory then we 
+#     assume the patch is there...
+#   - we parse the install inventory as this is far faster than
+#     using opatch or weblogic.version
+hasWebLogicPatches()
+{
+  local reg_file=$ORACLE_HOME/inventory/registry.xml
+  [ ! -f $reg_file ] && return 0
+  for pnum in "$@"; do
+    grep --silent "patch-id=\"$1\"" $reg_file || return 1
+  done
+}
+
+# getWebLogicVersion
+#   parse wl version from install inventory
+#   - if we can't get a version number then we return
+#     a high dummy version number that's sufficient
+#     to pass version checks "9999.9999.9999.9999"
+#   - we parse the install inventory as this is far faster than
+#     using opatch or weblogic.version
+getWebLogicVersion()
+{
+  local reg_file=$ORACLE_HOME/inventory/registry.xml
+
+  [ ! -f $reg_file ] && echo "9999.9999.9999.9999" && return
+
+  local wlver="`grep 'name="WebLogic Server".*version=' $reg_file \
```

Comment:
The registry.xml entry in FMW image ( container-registry.oracle.com/middleware/fmw-inf:12.2.1.3) and it is different: 
      <distribution status="installed" name="WebLogic Server for FMW" version="12.2.1.3.0">

Reply:
Ah - you're right.  It adds ' for FMW'!  Good catch!  I've modified the grep expression from `name="WebLogic Server".*` to `name="WebLogic Server.*`.

Reply:
I also ran a little test with the FMW image to see if the version was obtained - it was...